src/s3/client/paginated.rs

```diff
@@ -0,0 +1,96 @@
+use crate::s3::{
+    args::ListObjectsV2Args, client::Client, error::Error, response::ListObjectsV2Response,
+    utils::Multimap,
+};
+
+use futures_core::Stream;
+use futures_util::stream as futures_stream;
+
+#[derive(Debug, Clone)]
+pub struct ListObjectsV2ArgsOwned {
+    extra_headers: Option<Multimap>,
+    extra_query_params: Option<Multimap>,
+    region: Option<String>,
+    bucket: String,
+    delimiter: Option<String>,
+    encoding_type: Option<String>,
+    max_keys: Option<u16>,
+    prefix: Option<String>,
+    start_after: Option<String>,
+    continuation_token: Option<String>,
+    fetch_owner: bool,
+    include_user_metadata: bool,
+}
```

Comment:
We can avoid this extra struct by removing the lifetime parameter from `ListObjectsV2Args<'a>`. Because of references in the struct, it becomes complicated to make a stream out of it.

Reply:
Apart from higher level listing, we also provide lower level S3 APIs to be used by advanced users. As long as it doesn't break the user experience, I am good.

I am fine by having streaming support as a addition to current callback functionality. At the time of writing I evaluated both the approaches and chose callback approach because of the nature of List Objects S3 APIs. As we have result in hand as vector, the vector can be streamed. If someone wants streaming, they could write it on top level this vector.